src/core/packet.lua

```diff
@@ -109,3 +114,21 @@ function preallocate_step()
    packet_allocation_step = 2 * packet_allocation_step
 end
 
+function dump(p, w)
+   w = w or io.write
+   for i = 0, p.length-1 do
+      if i % 16 == 0 then w('\n', bit.tohex(i, -4), ': ') end
+      w(bit.tohex(p.data[i], -2), ' ')
+   end
+   w('\n')
+end
+
+ffi.metatype(packet_t, {__index = {
+   clone = clone,
+   append = append,
+   prepend = prepend,
+   shiftleft = shiftleft,
+   free = free,
+   physical = physical,
+   dump = dump,
+}})
```

Comment:
Is it important to add an FFI metatable to the packet struct in this PR? This looks like an unrelated change that is coming along for the ride.

I do actually think that it is a good time for us to define a unified "Snabb API" and that FFI metatables are probably the right way to dispatch calls to packets, links, etc (instead of full module calls that can be slow and local caching of functions that is verbose). However I am not so wild about doing this piecemeal unless this is solving an important problem somehow.